config_examples/template_config.yaml

```diff
@@ -45,24 +43,94 @@ telescope:
         sep: 14.6
         split_core: True
         outriggers: 2
+    # TODO: check how pyuvsim handles beams, see if YAML constructor needs
+    # to be updated/modified.
+    omega_p: !Beam
+        # TODO: add support for specifying directory containing beam file
+        datafile: HERA_H2C_BEAM_MODEL.npz
+        interp_kwargs:
+            interpolator: interp1d
+            fill_value: extrapolate
+            # if you want to use a polynomial interpolator instead, then
+            # interpolator: poly1d
+            # kwargs not accepted for this; see numpy.poly1d documentation
+    Trx: 0.0 # receiver temperature for thermal noise
+defaults:
+    # behavior of specifying default config and some other
+    # parameters (e.g. omega_p, Tsky_mdl) currently unknown
+    default_config: 'h2c'
 systematics:
-    # TODO: fill out what should go in here
-analysis:
-    # TODO: fill out what should go in here
-simulation:
+    # XXX: perhaps change include section of simulation to just list rfi
+    # and include all the specified rfi effects if rfi is listed
+    rfi:
+        # see hera_sim.rfi documentation for details on parameter names
+        rfi_stations:
+            stations: !!null
+        rfi_impulse:
+            chance: 0.001
+            strength: 20.0
+        rfi_scatter:
+            chance: 0.0001
+            strength: 10.0
+            std: 10.0
+        rfi_dtv:
+            freq_min: 0.174
+            freq_max: 0.214
+            width: 0.008
+            chance: 0.0001
+            strength: 10.0
+            strength_std: 10.0
+    sigchain:
+        gains:
+            gain_spread: 0.1
+            dly_rng: (-20, 20)
+        # XXX: the gen_reflection_gains function is somewhat difficult
```

Comment:
the problem is that reflection gains requires a list for parameters like `amp` that has the same length as the number of antennas, and it defaults to zero if a list is not passed. from the docstring:
```
        amp (list, optional): antenna reflection amplitudes for each antenna. Default is 1.0
        dly (list, optional): antenna reflection delays [nanosec]. Default is 0.0
        phs (lists, optional): antenna reflection phases [radian]. Default is 0.0
```
it says the default amplitude is 1.0, but it's actually 0.0

Reply:
Hmmm, yeah array-valued arguments are always going to be tricky. One way I've tried to deal with this before is to have a constructor that basically says "this is going to be a file... read it in". You can even go so far as to have a _list_ of possible ways to read it in (i.e. from `npz`, `npy`, HDF5, ASCII,...), and cycle through them until something works.

Reply:
Okay, I suppose we could require that this part be dealt with by specifying the parameters in a file, but I don't think we should just blindly guess at which way the file has been formatted. I think, if we're going to use a parameter file here, that it is best to explicitly state which file format is expected and how the file contents should be arranged.

I've been thinking that it would be a good idea to update the `gen_reflection_gains` function to allow for keywords to be used to make the lists (e.g. 'random' for the phases would choose all of the phases randomly), but I think this is not the place to have that discussion.

Ultimately, I'm not convinced that we absolutely need to solve this issue before merging. The main reason this PR was even made was to provide an interface for quickly generating a mock set of data to be used for benchmarking tests, and I suspect that there isn't a huge demand for including cable reflections in the mock data. So for now, I'd say that anyone who really wants to use this feature should just input the values as a list (no problem doing this with a YAML), and we can work on developing a nicer interface in the future.

Reply:
OK yes I agree -- we can decide on a standard and stick to it. I also agree that it doesn't *need* to be done right now. Let's get this into the wild, and see how it works in practice, and refine from there. In any case, input as a list (in YAML) is something we probably want to support, even if other better methods are also supported.